src/toil/cwl/cwltoil.py

```diff
@@ -753,17 +757,42 @@ def size(self, path: str) -> int:
             return super().size(path)
 
 
+def write_to_pipe(file_store: AbstractFileStore, pipe_name: str, id: FileID) -> None:
+    with file_store.readGlobalFileStream(id) as fi:
+        with open(pipe_name, 'wb') as fifo:
+            chunkSz = 1024 * 1024
+            while True:
+                data = fi.read(chunkSz)
+                if not data:
+                    break
+                fifo.write(data)
+
+
 def toil_get_file(
-    file_store: AbstractFileStore, index: dict, existing: dict, file_store_id: str
+    file_store: AbstractFileStore, index: dict, existing: dict, file_store_id: str, streamable: bool,
+        streaming_allowed: bool
 ) -> str:
     """Get path to input file from Toil jobstore."""
     if not file_store_id.startswith("toilfs:"):
         return file_store.jobStore.getPublicUrl(
             file_store.jobStore.importFile(file_store_id, symlink=True)
         )
-    src_path = file_store.readGlobalFile(FileID.unpack(file_store_id[7:]), symlink=True)
+
+    file_id = FileID.unpack(file_store_id[7:])
+
+    if streaming_allowed and streamable and isinstance(file_store, NonCachingFileStore) and \
```

Comment:
When the `file_store` is a `CachingFileStore` then `file_id` is a SQLite object and `SQLite objects created in a thread can only be used in that same thread`

Reply:
@DailyDreaming Any thoughts here?

Reply:
This is not correct, I think that the error is from connection parameter from `CachingFileStore`. I have an idea how to fix this